frontend/src/components/NewPassword/NewPassword.js

```diff
@@ -22,14 +27,34 @@ class NewPassword extends Component {
     this.setState({ [event.target.name]: event.target.value });
   };
 
+  handleSubmit = (event) => {
+    const { token } = this.props;
+    event.preventDefault();
+    const { password, confirmPassword } = this.state;
+    if (password !== confirmPassword) {
+      this.setState({ error: "Passwords don't match" });
+    } else {
+      resetPassword(token, password)
+        .then(this.handleRedirect)
+        .catch((error) => {
+          this.setState({ error: error.message });
```

Comment:
I think the error is actually going to be on something like `error.response.data.errors[0].message`. `error.message` is probably just going to be "the request failed with XXX status." I think we may want to add some handling here, like if the error is a 400-level error, it indicates a bad token or bad password.